app/summary.py

```diff
@@ -0,0 +1,50 @@
+import pickle
+import pandas as pd
+from datetime import datetime
+
+from jjit import search_jjit
+from pracuj import search_pracuj
+
+db_path = 'offers_db'
+
+def prepare_tech_dict():
+
+    offers_db = pickle.load(open(db_path, 'rb'))
+
+    tech_dict = {}
+
+    for offer_list in offers_db['technologies']:
+        for tech in offer_list:
+            if tech in tech_dict:
+                tech_dict[tech] += 1
+            else:
+                tech_dict[tech] = 1
+
+    with open('tech_dict', 'wb') as file:
+        pickle.dump(tech_dict, file)
+
+def search_offers(pracuj_url, jjit_categories, jjit_experience):
+    offers_jjit = search_jjit(jjit_categories, jjit_experience)
+    offers_pracuj = search_pracuj(pracuj_url)
+
+    new_offers = pd.concat([offers_jjit, offers_pracuj])
+    new_offers = new_offers.drop_duplicates(subset=new_offers.columns[:-2])
```

Comment:
Nie jestem pewien czy concat tych dwóch df-ów jest dobrym pomysłem, bo na różnych portalach modele danych (czyli np. to w jaki sposób zostały sparametryzowane oferty) mogą być różne. Domyślam się, że jeśli to działa to zostało to jakoś znormalizowane, ale wciąż traktowałbym przeszukiwanie tych dwóch portali zupełnie niezależnie. Podobnie jak w moim projekcie z nieruchomościami Otodom i Domiporta są całkowicie odseparowane

Reply:
Co do subsetu w `drop_duplicates`: patrząc na ten kod nie wiem jakie to są te dwie ostatnie kolumny. W pandasie kiedy możemy to raczej używamy nazw kolumn a nie tak jak w numpy, gdzie nazw nie ma.